components/brave_federated/adapters/flower_helper.cc

```diff
@@ -0,0 +1,185 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated/adapters/flower_helper.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/logging.h"
+#include "brave/components/brave_federated/task/model.h"
+#include "brave/third_party/flower/src/brave/flwr/serde.h"
+#include "brave/third_party/flower/src/proto/flwr/proto/fleet.pb.h"
+#include "brave/third_party/flower/src/proto/flwr/proto/node.pb.h"
+#include "brave/third_party/flower/src/proto/flwr/proto/transport.pb.h"
+
+namespace brave_federated {
+
+std::string BuildGetTasksPayload() {
+  flower::Node node;
+  node.set_node_id(0);
+  node.set_anonymous(true);
+
+  flower::PullTaskInsRequest pull_task_instructions_request;
+  *pull_task_instructions_request.mutable_node() = node;
+  pull_task_instructions_request.add_task_ids("0");
+
+  std::string request;
+  pull_task_instructions_request.SerializeToString(&request);
```

Comment:
Can `SerializeToString` fail? If so, `request` would be empty, should we be handling this case?

Reply:
`SerializeToString` returns a bool. We should handle this case. Maybe return as `absl::optional<std::string>` and handle these failure cases.

Reply:
opted for `base::expected` for consistency. Is that okay?

Reply:
I think absl::optional makes more sense here, because you would still return an empty string using base::expected. Here we are saying that we dont have a value and therefore can check for absl::nullopt